dss_datamover/utils/utility.py

```diff
@@ -206,17 +206,34 @@ def get_s3_prefix(logger, nfs_cluster, prefix=None):
             yield nfs_server_ip + "/"
 
 
-def validate_s3_prefix(logger, prefix):
+def validate_s3_prefix(logger, prefix, config_nfs=None):
     """
     Validate a given prefix. A S3 prefix should start without "/" and end with "/".
     <prefix string>/
     :param logger: multiprocessing logger object
     :param prefix: a string
+    :param(optional) config_nfs: nfs_share Config dict
     :return: Success/Failure
     """
+    inv_prefix = 0
     if prefix.startswith("/") or not prefix.endswith("/"):
-        logger.error("WRONG specification of prefix. Should be in the format of <nfs_server_ip>/<prefix>/ ")
+        logger.fatal("WRONG specification of prefix. Should be in the format of <nfs_server_ip>/<prefix>/ ")
         return False
+    if config_nfs is not None:
+        cluster_ip = prefix.split('/')[0]
+        if config_nfs and cluster_ip in config_nfs:
```

Comment:
You already checked `config_nfs` on the top. Why to check again?

Reply:
This was actually a double validation...
the 1st clause was to check whether the parameter value was being passed by the caller and the 2nd clause was to check whether the passed value was an empty dict....

Anyways, it did seem clogged up, so have updated the logic in my latest commit.